src/shared/telemetry/activation.ts

```diff
@@ -72,24 +75,28 @@ export async function activate(activateArguments: {
         activateArguments.extensionContext.subscriptions
     )
 }
-
-export function isTelemetryEnabled(toolkitSettings: SettingsConfiguration): boolean {
+/*
+ * Formats the AWS telemetry setting to a boolean: false if setting value was 'Disable' or false, true for everything else
+ */
+export async function formatTelemetrySettingToBool(toolkitSettings: SettingsConfiguration): Promise<void> {
     // Setting used to be an enum, but is now a boolean.
     // We don't have api-based strong type support, so we have to process this value manually.
     const value = toolkitSettings.readSetting<any>(AWS_TELEMETRY_KEY)
 
-    // Handle original opt-out value (setting used to be a tri-state string value)
-    if (value === LEGACY_SETTINGS_TELEMETRY_VALUE_DISABLE) {
-        return false
-    }
+    if (typeof value === 'boolean') return
 
-    // Current value is expected to be a boolean
-    if (typeof value === 'boolean') {
-        return value
+    // Original setting used to be a tri-state string value
+    // Set to false if original setting was opt-out
+    // Treat anything else (unexpected values, datatypes, or undefined) as opt-in
+    if (value === LEGACY_SETTINGS_TELEMETRY_VALUE_DISABLE) {
+        await toolkitSettings.writeSetting<any>(AWS_TELEMETRY_KEY, false, vscode.ConfigurationTarget.Global)
+    } else {
+        await toolkitSettings.writeSetting<any>(AWS_TELEMETRY_KEY, true, vscode.ConfigurationTarget.Global)
     }
+}
 
-    // Treat anything else (unexpected values, datatypes, or undefined) as opt-in
-    return true
+export function isTelemetryEnabled(toolkitSettings: SettingsConfiguration): boolean {
+    return toolkitSettings.readSetting<any>(AWS_TELEMETRY_KEY)
```

Comment:
The value _should_ be sanitized, however one could still go into the file and put any value the like here. It may be worth retaining the previous function implementation to prevent calling code from having unexpected data.

Reply:
Great point, I'll restore the functionality.

Reply:
This could also be true of any value we read from the config, since VSCode doesn't throw an exception if it violates the schema. Perhaps we should either

1. Throw an error if the value is unexpected
2. Default to turning off the telemetry if the value is unexpected

Reply:
I prefer 1 since the point of this method is to query, not to modify